src/Terminal.js

```diff
@@ -954,8 +954,38 @@ var Terminal = {
                 }
 				break;
 			case "top":
-				//TODO List each's script RAM usage
-                post("Not yet implemented");
+                if(commandArray.length != 1) {
+                  post("Incorrect usage of top command. Usage: top"); return;
+                }
+                
+                post("Script                    RAM Usage                 Threads");
+                var currRunningScripts = Player.getCurrentServer().runningScripts;
+                for(i = 0; i < currRunningScripts.length; i++) {
+                    //Check if script is on current server
+                    for(j = 0; j < Player.getCurrentServer().scripts.length; j++) {
+                        if(currRunningScripts[i] === Player.getCurrentServer().scripts[j].filename) {
```

Comment:
The first loop iterates through the current running scripts on the server, while the second iterates through the scripts property of the server, which has the ram usage for each script (That was the only way I could find to determine it). The if statement only passes if the script is running on the current server.